newm-server/src/main/kotlin/io/newm/server/features/earnings/repo/EarningsRepositoryImpl.kt

```diff
@@ -312,4 +323,62 @@ class EarningsRepositoryImpl(
 
         schedulerDaemon.scheduleJob(jobDetail, trigger)
     }
+
+    override suspend fun generateRetrieveEarningsPaymentTransaction(
+        songId: SongId,
+        requesterId: UserId,
+        sourceUtxos: List<Utxo>,
+        changeAddress: String
+    ): String {
+        log.debug { "generateRetrieveEarningsPaymentTransaction: songId = $songId" }
+
+        checkRequester(songId, requesterId, verified = true)
+
+        val song = get(songId)
+        val key = Key.generateNew()
+        val keyId = cardanoRepository.saveKey(key)
+        val transaction =
+            cardanoRepository.buildTransaction {
+                this.sourceUtxos.addAll(sourceUtxos)
+                this.outputUtxos.add(
+                    outputUtxo {
+                        address = key.address
+                        lovelace = song.mintCostLovelace.toString()
```

Comment:
lovelace should come from the config database "earnings.claimOrderFee" or whatever constant we have for that in the code.